lib/puppetserver/ca/action/list.rb

```diff
@@ -78,8 +87,17 @@ def run(input)
           all_certs = get_all_certs(puppet.settings).select { |cert| filter_names.call(cert) }
           requested, signed, revoked = separate_certs(all_certs)
           missing = certnames - all_certs.map { |cert| cert['name'] }
+          
+          if all && output_format == 'json'
+            grouped_cert = { "requested" => requested,
+                             "signed" => signed, 
+                             "revoked" => revoked,
+                             "missing" => missing }.to_json
+            puts grouped_cert
+            return 0 
+          end
 
-          (all || certnames.any?) \
+          ((all || certnames.any?) && output_format == nil) \
```

Comment:
I put the `nil` check there because the previous check on line 65 only take in account that the user enter some sort of format.  But make sense since I was returning in the block right above it.

Reply:
yep!